rust_src/src/xdisp.rs

```diff
@@ -0,0 +1,22 @@
+use lisp::LispObject;
+use threads::ThreadState;
+use remacs_macros::lisp_fn;
+use remacs_sys::bset_update_mode_line;
+use libc::c_void;
+
+/// Mark the current buffer for redisplay.
+/// This function may be passed to `add-variable-watcher`.
+#[lisp_fn]
+fn set_buffer_redisplay(
+    _symbol: LispObject,
+    _newval: LispObject,
+    _op: LispObject,
+    _where: LispObject,
+) -> LispObject {
+    unsafe {
+        bset_update_mode_line(ThreadState::current_buffer().as_ptr() as *const _ as
```

Comment:
With the change above, this can use `as_mut()` and not need a cast.

Reply:
consider it done. I had used as_mut(), but I didn't get the signature right so of course it wouldn't compile. Thanks.